### PR TITLE
Bug 1216896 - Localize the keyboard layout label "Emoji"

### PR DESCRIPTION
**Pull request by timdream**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=6ed5117307bfe2e2c5054accac9b2adb42a36fea

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d2a5a8d6eeaf26a091df0bcbf9f8b545acf11487

File: apps/settings/locales/settings.en-US.properties [hunk 1]
```diff
@@ -770,6 +770,10 @@ keyboardType-url=url
 keyboardType-number=number
 defaultKeyboardEnabled={{appName}} {{layoutName}} is set as default input method
 
+# LOCALIZATION NOTE (layout-emoji): Label for the layout "Emoji".
+# other layouts are always labeled in their own native scripts (in JS files)
+keyboard-layout-emoji=Emoji
```
Comment (gasolin) [inline apps/settings/locales/settings.en-US.properties]: \o/

File: apps/settings/js/modules/keyboard_context.js [hunk 2]
```diff
@@ -76,11 +78,13 @@ define(function(require) {
    * @returns {Layout}
    */
   var Layout =
-    function(id, appName, appManifestURL, name, description, types, enabled) {
+    function(id, appName, appManifestURL, name, nameL10nId, description,
+            types, enabled) {
```
Comment (jedireza) [inline apps/settings/js/modules/keyboard_context.js]: minor nit: might be good to pass an object instead

Comment (timdream) [inline apps/settings/js/modules/keyboard_context.js]: I would defer that to @gasolin

Comment (gasolin) [inline apps/settings/js/modules/keyboard_context.js]: ya, it will be nice to pass with object instead

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=73d3ebd63b80146c98588bb6d3c7328639756961

File: apps/settings/js/modules/keyboard_context.js [hunk 3]
```diff
@@ -130,9 +134,15 @@ define(function(require) {
         app[layout.layoutId].enabled = layout.enabled;
         return app[layout.layoutId];
       }
+
+      var layoutName = layout.inputManifest.nameL10nId ?
+        navigator.mozL10n.get(layout.inputManifest.nameL10nId) :
```
Comment (albertopq) [inline apps/settings/js/modules/keyboard_context.js]: We should stop using mozL10n.get:
https://developer.mozilla.org/en-US/Apps/Build/Localization/Localization_code_best_practices#Do_not_use_mozL10n.get

File: apps/settings/js/modules/keyboard_context.js [hunk 4]
```diff
@@ -193,6 +203,9 @@ define(function(require) {
         keyboard.description = keyboardManifest.description;
         keyboard.layouts.forEach(function(layout) {
           layout.appName = keyboardManifest.name;
+          if (layout.nameL10nId) {
+            layout.name = navigator.mozL10n.get(layout.nameL10nId);
```
Comment (albertopq) [inline apps/settings/js/modules/keyboard_context.js]: Same here

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1ca5cc1417e0776ef9c59955024ec835985a5c15

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2782ad7bcbbc42809e381287d4e6eb6b91469b5e

File: apps/system/js/input_layouts.js [hunk 5]
```diff
@@ -85,20 +85,22 @@
...
-    // define properties for name that resolve at display time
-    // to the correct language via the ManifestHelper
+    // define properties for names that resolve at display time
+    // to the correct language via the ManifestHelper or mozL10n.get()
     Object.defineProperties(transformedLayout, {
       name: {
-        get: getName.bind(layout.inputManifest),
+        get: function() {
+          if (layout.inputManifest.nameL10nId) {
+            return navigator.mozL10n.get(layout.inputManifest.nameL10nId);
```
Comment (albertopq) [inline apps/system/js/input_layouts.js]: Still mozL10n.get here. If you need to dinamically retrieve it, use formatValue instead

**Comment by timdream** [comment]
To be merged in #33540

**Pull request closed without merging**